crypto/include/internal/cryptlib.h

```diff
@@ -74,6 +74,8 @@ FILE *openssl_fopen(const char *filename, const char *mode);
 void *openssl_fopen(const char *filename, const char *mode);
 # endif
 
+unsigned long OPENSSL_rdtsc(void);
```

Comment:
Wasn't it decided in #4757 after a lengthy discussion that this should consistently be a uint32_t?  (see db26a403b4de69d7846656362e70f5083428)

Reply:
correction: d807db26a403b4de69d7846656362e70f5083428

Reply:
Maybe as another PR. This PR is just trying the fix the --strict-warnings build.
There are likely more issues, for instance sparcv9cap.c
defines "unsigned long OPENSSL_rdtsc(void)" without including "internal/cryptlib.h" first.